app/src/main/java/com/android/solvit/shared/model/request/ServiceRequestRepositoryFirebase.kt

```diff
@@ -116,13 +151,17 @@ class ServiceRequestRepositoryFirebase(
           title = document.getString("title") ?: "",
           description = document.getString("description") ?: "",
           userId = document.getString("userId") ?: "",
+          providerId = document.getString("providerId"),
           dueDate = document.getTimestamp("dueDate") ?: Timestamp.now(),
+          meetingDate = document.getTimestamp("meetingDate"),
           location =
               Location(
                   latitude = document.getDouble("location.latitude") ?: 0.0,
                   longitude = document.getDouble("location.longitude") ?: 0.0,
                   name = document.getString("location.name") ?: ""),
           imageUrl = document.getString("imageUrl"),
+          packageId = document.getString("packageId"),
```

Comment:
We should take into account that not every provider propose packages ! Maybe if a provider doesn't offer this service, then trying to get the package id from the doc that will lead to an error! We can add a default value for example 0 and with this value we know that he has not packages proposals!

Reply:
You are right, however the translation from the document to the Service Request will automatically take care of null items, returning a null value for the missing parameter, which we can afterwards check for when manipulating service requests.